File: tyama_icpc2012dA.nim
#[
/usr/bin/env nim c --nimcache:$HOME/.nimcache --opt:speed --run $0 $@;exit
#]#
import strutils
var n=parseInt(readLine(stdin))
for i in 1..n:
 var YMD=readLine(stdin).split()
 var y=parseInt(YMD[0])-1
 var m=parseInt(YMD[1])-1
 var d=parseInt(YMD[2])
 var x=0
 if y mod 3<2:
  x=m div 2
 echo(196471-y*195-y div 3*5-m*20+x-d)

File: tyama_icpc2012dA.kt
fun main(args: Array<String>){
	val n=readLine()!!.toInt()
	for(i in 0..n-1){
		var (y,m,d)=readLine()!!.split(" ").map(String::toInt)
		y-=1
		m-=1
		var r=0
		if(y%3!=2)r=m/2
		println(196471-y*195-y/3*5-m*20+r-d)
	}
}
